src/Symfony/Component/HttpClient/HttpClientTrait.php

```diff
@@ -438,7 +446,7 @@ private static function removeDotSegments(string $path)
         $result = '';
 
         while (!\in_array($path, ['', '.', '..'], true)) {
-            if ('.' === $path[0] && (0 === strpos($path, $p = '../') || 0 === strpos($path, $p = './'))) {
+            if (0 === strpos($path, '.') && (0 === strpos($path, $p = '../') || 0 === strpos($path, $p = './'))) {
```

Comment:
This seams to be a bit slower, right?

Reply:
right, this should be reverted

Reply:
It is a bit slower indeed.

I did it for two reasons:
1) For consistency
2) In the future, treating string as array won't work anymore.

See: https://3v4l.org/368ZB

Reply:
consistency doesn't apply here, and the future is not known beforehand :)

Reply:
Okay, I will revert it.

FYI, PHP 7.3 print a warnings when treating strings are array.

See the example: https://3v4l.org/368ZB

Reply:
Sorry for the confusion. Using `strpos()` with non string arguments causes the `deprecated warning,` not using string as arrays. Reverting this.

Reply:
>FYI, PHP 7.3 print a warnings when treating strings are array.

Where did you see that? On my computer:

```
php > echo PHP_VERSION;
7.3.3-1+ubuntu18.04.1+deb.sury.org+1
php > dump('foobar'[2]);
"o"
```

Reply:
Sorry, my explanation was wrong.
I wrote this message because my example had a typo in it. So I though this was the issue.
Sorry again for the noise and please discard this.